src/open_inwoner/react/modules/KVKBranchSelectorModule/KVKBranchSelectorModule.tsx

```diff
@@ -0,0 +1,82 @@
+import { ReactElement } from 'react'
+import { createRoot, Root } from 'react-dom/client'
+import { IntlProvider } from 'react-intl'
+import { KVKBranchSelector } from '@react/components/KVKBranchSelector'
+import { getIntlProviderProps } from '@react/i18n/i18n'
+import { getJsonFromScriptTag } from '@react/lib/getJsonScriptData'
+
+export interface ComboBoxItem {
+  id: string
+  label: string
+  subLabel?: string
+  vestigingsnummer?: string
+  type?: string
+}
+
+/**
+ * Module for mounting KVKBranchSelector React components into Django templates.
+ * Searches for elements with data-react-module="kvkbranchselector" and initializes them.
+ */
+export default class KVKBranchSelectorModule {
+  static async init(): Promise<void> {
+    const intlProps = await getIntlProviderProps()
+    const nodes = document.querySelectorAll<HTMLElement>(
+      '[data-react-module="kvkbranchselector"]'
+    )
+
+    nodes.forEach((el) => {
+      try {
+        // Reuse existing React root if available to prevent remounting
+        const rootKey = '_reactRoot'
+        let root: Root | undefined = (el as any)[rootKey]
+
+        // Extract branch data from embedded JSON script tag
+        const data = getJsonFromScriptTag<{
+          items: ComboBoxItem[]
+          selected_id?: string
+        }>('branch-data')
+
+        if (!data || !data.items || data.items.length === 0) {
+          console.error('[KVKBranchSelectorModule] No branches available')
```

Comment:
I usually add `new Error`, bc this gives some context about the stack trace.

```suggestion
          console.error(new Error('[KVKBranchSelectorModule] No branches available'))
```

Reply:
Nice.